test/src/api/commands/element/testSetPasswordCheckReports.js

```diff
@@ -0,0 +1,80 @@
+const assert = require('assert');
+const path = require('path');
+const {Key} = require('selenium-webdriver');
+const common = require('../../../../common.js');
+const {settings} = common;
+const NightwatchClient = common.require('index.js');
+const MockServer  = require('../../../../lib/mockserver.js');
+const CommandGlobals = require('../../../../lib/globals/commands.js');
+
+describe('setPassword check', function() {
+
+  before(function(done) {
+    this.server = MockServer.init();
+    this.server.on('listening', () => done());
+  });
+
+  after(function(done) {
+    this.server.close(function() {
+      done();
+    });
+  });
+
+  it('client.setPassword() value redacted in rawHttpOutput', async function() {
+
+    MockServer.addMock({
+      url: '/wd/hub/session/1352110219202/element/0/value',
+      method: 'POST',
+      postdata: {
+        text: Key.NULL + 'password',
+        value: [Key.NULL, 'p', 'a', 's', 's', 'w', 'o', 'r', 'd']
+      },
+      response: {
+        sessionId: '1352110219202',
+        status: 0
+      }
+    });
+
+    MockServer.addMock({
+      url: '/wd/hub/session/1352110219202/element/0/clear',
+      method: 'POST',
+      response: {
+        sessionId: '1352110219202',
+        status: 0
+      }
+    });
+
+    const testsPath = [
+      path.join(__dirname, '../../../../sampletests/checkValueRedacted/passwordValueRedacted.js')
+    ];
+
+    const globals = {
+      calls: 0,
+      waitForConditionTimeout: 10,
+      waitForConditionPollInterval: 10,
+      reporter(results) {
```

Comment:
Similarly, we need to make sure that the `reporter` function is actually called. We can have a similar `globalReporterCalled` variable at the top, set it in the `reporter` function and assert it at the end of the test.

Reply:
added this assertion [9ad5d68](https://github.com/nightwatchjs/nightwatch/pull/4066/commits/9ad5d68ae6a1aa07a9b487d389e95de41976f225)